src/components/newsletter/newsletter-sign-up.tsx

```diff
@@ -0,0 +1,264 @@
+    line-height: 2.75rem;
+    font-weight: 600;
+    color: var(--midnight);
+    margin-bottom: 0.75rem;
+  }
+
+  p {
+    line-height: 1.875rem;
+    color: var(--midnight);
+  }
+
+  form {
+    display: block;
+  }
+
+  label {
+    display: block;
+    font-weight: 600;
+    line-height: 2.125rem;
+    color: var(--midnight);
+  }
+
+  fieldset {
+    display: contents;
+  }
+
+  input[type="text"],
+  input[type="email"] {
+    border: 1px solid var(--midnight);
+    border-radius: 3px;
+    width: 100%;
+    height: 3rem;
+    padding: 7px 18px 7px 10px;
+
+    &:focus {
+      outline: none;
+      border-color: var(--cobalt);
+    }
+  }
+
+  input[type="submit"] {
+    width: 100%;
+    height: 3rem;
+    background-color: var(--midnight);
+    color: var(--white);
+    line-height: 2.125rem;
+    font-weight: 700;
+
+    &:hover {
+      background-color: var(--denim);
+      cursor: pointer;
+    }
+  }
+
+  ${mq(BreakPoint.md)} {
+    padding-left: 5rem;
+    padding-right: 5rem;
+    input[type="text"],
+    input[type="email"] {
+      width: 50%;
+    }
+    input[type="submit"] {
+      width: 40%;
+    }
+  }
+
+  ${mq(BreakPoint.lg)} {
+    padding-left: 10rem;
+    padding-right: 10rem;
+
+    form {
+      display: grid;
+      grid-template-columns: 1fr 1fr 1fr;
+      column-gap: 1.875rem;
+    }
+
+    input[type="text"],
+    input[type="email"] {
+      width: 100%;
+    }
+    input[type="submit"] {
+      width: 100%;
+    }
+  }
+
+  .full-span {
+    grid-column: 1 / span 3;
+    text-align: left;
+    margin-top: 1.375rem;
+  }
+
+  a {
+    text-decoration: underline;
+    font-weight: 600;
+  }
+
+  input[type="text"].error,
+  input[type="email"].error {
+    border-color: var(--error);
+  }
+
+  .input-error {
+    color: var(--error);
+    line-height: 1.625rem;
+    margin-top: 0.75rem;
+    display: none;
+  }
+
+  input[type="text"].error + .input-error,
+  input[type="email"].error + .input-error {
+    display: block;
+  }
+
+  p.response {
+    display: none;
+    margin-bottom: 1rem;
+  }
+
+  p.response.success {
+    display: block;
+    color: var(--midnight);
+  }
+
+  p.response.error {
+    display: block;
+    color: var(--error);
+  }
+
+  .submitSection {
+    label {
+      width: 1px;
+    }
+  }
+`;
+
+type SignUpStatus = {
+  success: boolean;
+  msg: string;
+};
+
+const NewsletterSignUp = () => {
+  const [email, setEmail] = useState(undefined);
+  const [emailError, setEmailError] = useState(false);
+  const [name, setName] = useState(undefined);
+  const [nameError, setNameError] = useState(false);
+  const [mcResponse, setMcResponse] = useState<SignUpStatus | undefined>(
+    undefined
+  );
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+
+    setMcResponse(undefined);
+
+    if (!name) setNameError(true);
+    if (!email) setEmailError(true);
+    if (!name || !email) return;
+
+    const response = await addToMailchimp(email, { FNAME: name });
```

Comment:
I've got some pretty heavy user-tracking settings on Firefox, and it blocked the call here. There's no error handling which covers the network request being blocked, so the page errored for me. We should handle the case where the call to `addToMailChimp` raises a JavaScript error.

Reply:
good shout 👍